src/main/java/org/opensearch/security/authtoken/jwt/JwtVendor.java

```diff
@@ -11,63 +11,43 @@
 
 package org.opensearch.security.authtoken.jwt;
 
-import java.io.IOException;
-import java.security.AccessController;
-import java.security.PrivilegedAction;
 import java.text.ParseException;
-import java.util.ArrayList;
 import java.util.Base64;
-import java.util.Date;
 import java.util.List;
-import java.util.Optional;
-import java.util.function.LongSupplier;
-
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
 
 import org.opensearch.OpenSearchException;
 import org.opensearch.common.collect.Tuple;
 import org.opensearch.common.settings.Settings;
-import org.opensearch.common.xcontent.XContentFactory;
-import org.opensearch.core.xcontent.ToXContent;
-import org.opensearch.security.action.apitokens.ApiToken;
 
 import com.nimbusds.jose.JOSEException;
 import com.nimbusds.jose.JWSAlgorithm;
-import com.nimbusds.jose.JWSHeader;
 import com.nimbusds.jose.JWSSigner;
 import com.nimbusds.jose.KeyLengthException;
 import com.nimbusds.jose.crypto.MACSigner;
 import com.nimbusds.jose.jwk.JWK;
 import com.nimbusds.jose.jwk.KeyUse;
 import com.nimbusds.jose.jwk.OctetSequenceKey;
-import com.nimbusds.jwt.JWTClaimsSet;
-import com.nimbusds.jwt.SignedJWT;
 
 import static org.opensearch.security.util.AuthTokenUtils.isKeyNull;
 
-public class JwtVendor {
-    private static final Logger logger = LogManager.getLogger(JwtVendor.class);
-
-    private final JWK signingKey;
-    private final JWSSigner signer;
-    private final LongSupplier timeProvider;
-    private final EncryptionDecryptionUtil encryptionDecryptionUtil;
-    private static final Integer MAX_EXPIRY_SECONDS = 600;
-
-    public JwtVendor(final Settings settings, final Optional<LongSupplier> timeProvider) {
-        final Tuple<JWK, JWSSigner> tuple = createJwkFromSettings(settings);
-        signingKey = tuple.v1();
-        signer = tuple.v2();
-
-        if (isKeyNull(settings, "encryption_key")) {
-            throw new IllegalArgumentException("encryption_key cannot be null");
-        } else {
-            this.encryptionDecryptionUtil = new EncryptionDecryptionUtil(settings.get("encryption_key"));
-        }
-        this.timeProvider = timeProvider.orElse(System::currentTimeMillis);
+public abstract class JwtVendor {
+    public ExpiringBearerAuthToken createJwt(
+        final String issuer,
```

Comment:
I think we should work towards a single `JwtVendor` that can be initialized with Settings and separate claims builders for payloads for different types of tokens that this repo may issue.

i.e. `OBOClaims.Builder().issuer(<issuer>)` or `ApiTokenClaims.Builder()...`

Reply:
Agree. The reason why I had to split it up here was the Obo jwt vendor requires an encryption key, whereas my api token jwt vendor does not use the encryption key at all... what do you think is a good approach to centralize them when one has a required setting (errors out without it), whereas one does not use it?